apps/dashboards/tests/test_api.py

```diff
@@ -35,11 +40,22 @@ def test_get(self):
         response = self.client.get(url)
         assert response.status_code == 200
 
-    def test_get_all(self):
+    def test_list_view(self):
         url = reverse("visualization-list")
         response = self.client.get(url)
         assert response.status_code == 200
 
+    def test_list_view_as_non_staff(self):
+        request = self.factory.get(
+            reverse("visualization-list"),
+        )
+        request.user = self.user
+        request.user.is_staff = False
+        view = VisualizationViewSet.as_view(actions={"get": "list"})
+        force_authenticate(request, user=self.user)
+        response = view(request)
+        assert response.status_code == 200
```

Comment:
This is a step in the right direction but I would expect to see more detailed tests here. Add a test that actually shows that a non_staff user cannot access some visualizations or can only access certain visualizations.

Remember that the goal is to actually test the current implementation and not just to reach a coverage of 100%.